src/precice/impl/SolverInterfaceImpl.cpp

```diff
@@ -773,11 +779,54 @@ void SolverInterfaceImpl::setMeshQuad(
     PRECICE_CHECK(mesh->isValidEdgeID(secondEdgeID), "Given EdgeID is invalid!");
     PRECICE_CHECK(mesh->isValidEdgeID(thirdEdgeID), "Given EdgeID is invalid!");
     PRECICE_CHECK(mesh->isValidEdgeID(fourthEdgeID), "Given EdgeID is invalid!");
-    mesh::Edge &e0 = mesh->edges()[firstEdgeID];
-    mesh::Edge &e1 = mesh->edges()[secondEdgeID];
-    mesh::Edge &e2 = mesh->edges()[thirdEdgeID];
-    mesh::Edge &e3 = mesh->edges()[fourthEdgeID];
-    mesh->createQuad(e0, e1, e2, e3);
+    mesh::Edge *e[5];   //Contains the four edges of the quad and the new diagonal that will be created. 
+    e[0] = &mesh->edges()[firstEdgeID];
+    e[1] = &mesh->edges()[secondEdgeID];
+    e[2] = &mesh->edges()[thirdEdgeID];
+    e[3] = &mesh->edges()[fourthEdgeID];
+
+    PRECICE_CHECK(utils::unique_elements(utils::make_array(firstEdgeID, secondEdgeID, thirdEdgeID, fourthEdgeID)),
+                  "The four edge ID's are not unique. Please check that the edges that form the quad are correct.");
+
+    std::array<int,4>  edgeList;  // A list of edge ID numbers
```

Comment:
```suggestion
    std::array<int,4>  edgeIDs;  
```
more follow-up changes necessary below